status: update cmd ref

sanidhyamangal opened the pull request:
This pull request updates `\static\docs\commands-reference\status.md` for new options, usage examples and synopsis to fix #185.

~~@shcheklein  kindly check the changes, I missed optional arguments in the synopsis section so shall I add it in docs?~~

shcheklein commented on static/docs/commands-reference/status.md:
````diff
@@ -6,12 +6,15 @@ cache and local files, or between the local cache and remote cache.
 ```usage
-usage: dvc status [-h] [-q | -v] [-j JOBS] [--show-checksums]
-                  [-c] [-r REMOTE] [-a] [-T] [-d]
+usage: dvc status [-h] [-v] [-j JOBS] [--show-checksums] [-q] [-c] [-r REMOTE]
````
why this change? no need to be precise and copy it exactly from the DVC command line output

sanidhyamangal replied:
I did the same, simply copy-paste from command line output.

shcheklein replied:
_no_ need to copy it from the CLI output.

it makes sense to chage `[-q | -v]` but keep multiple lines to avoid scrolling. Especially on the mobile screen.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -6,12 +6,15 @@ cache and local files, or between the local cache and remote cache.
+                  [-a] [-T] [-d]
                   [targets [targets ...]]
 
+Show changed states, compare local cache and remote storage.
```
no need in this change, we don't include it usually. A summary at the very top of the page is enough. Check other command - like checkout fr example.

sanidhyamangal replied:
Okay, I will remove these changes.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -6,12 +6,15 @@ cache and local files, or between the local cache and remote cache.
+
 positional arguments:
-  targets               Limit the scope to these stage files.
+  targets               DVC files.
```
let's keep the previous version, and actually we need to update DVC itself. Can you create a patch for DVC repo with this simple change?

sanidhyamangal replied:
Okay, I will revert this to the original version. Okay sure will patch it in DVC as well.

sanidhyamangal replied:
@shcheklein  I have created a pull request for this in DVC kindly review it. 

https://github.com/iterative/dvc/pull/2111

sanidhyamangal commented on static/docs/commands-reference/status.md:
```diff
@@ -120,14 +120,14 @@ cache. For the typical process to update workspaces, see
-          outs
-                  changed:  foo
+bar.dvc:
+        changed deps:
```
@shcheklein, Check this I have already split the _changed_ into _deps_ and _outs_

sanidhyamangal commented:
> It looks great, thanks @sanidhyamangal ! A put a few comments to address. Among other things:
> 
> 1. There are a few new values for the `dvc status` in the local mode (w/o --cloud or --remote):
> 
> ```
>         if self.changed_md5():
>             ret.append("changed checksum")
> 
>         if self.is_callback:
>             ret.append("always changed")
> ```
> 
> description should be updated to include those. Please ask guys on dvc.org/chat #dev-general to explain those values.
> 
> 1. In the description _changed_ should be split now into _changed outs_ and _changed deps_.
> 2. Would be great to update the dvc status help message in the dvc repo to make it something like:
>    `An optional list of .dvc files to limit the scope. With -R accepts directories to search .dvc files in.`

@shcheklein  I couldn't get that last your last part so can you please throw some light on it.

sanidhyamangal commented:
@shcheklein  I am processing these changes still, need some to figure out exact functionality and reference that can be added into the description. If still, I have some problems I will address them to you on discord

sanidhyamangal commented:
@shcheklein, I have changed the description for the local mode as per your request so kindly review them. 

I have some questions regarding implementation `-R` flag, posted a query regarding same on discord kindly check that also.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -38,7 +38,10 @@ stages that affect the target stage.
 In the `local` mode, changes are detected through the checksum of every file
 listed in every stage file in the pipeline against the corresponding file in the
-file system. The output indicates the detected changes, if any. If no
+file system, using two modes `changed checksum` when actual checksum of data
```
these are not modes. These are different outputs/statuses. The paragraph that starts with "If instead differences ..." should be updated. Check the list of different outputs that starts line 57-60.

shcheklein commented:
@sanidhyamangal re the 3 - I meant more or less the change you made here https://github.com/iterative/dvc/pull/2111 .

sanidhyamangal commented:
@shcheklein, I have successfully updated the requested changes please go through them. I have also updated the changes from `iterative/dvc#2111` and created a pr for the same, so kindly go through it as well [iterative/dvc#2121](https://github.com/iterative/dvc/pull/2121)

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,12 @@ differ are listed. For each item listed, either the file name or the checksum is
+  - _changed checksum_ means actual check sum of data dosen't match specified in
+    dvc files
+  - _always checksum_ means special dvc file with no dependencies, considered
+    always changed
```
@efiop could you take a look please? do we mention all possible variants now?

efiop replied:
We are missing statuses for the out/dep files themselves. E.g. 
```
not in cache
deleted
modified
new
```

sanidhyamangal replied:
@efiop shall I add these statues too in the description?

efiop replied:
@sanidhyamangal Yes, please :slightly_smiling_face:

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,12 @@ differ are listed. For each item listed, either the file name or the checksum is
+  - _changed outs_ means the named file has changed outputs
+  - _changed checksum_ means actual check sum of data dosen't match specified in
+    dvc files
+  - _always checksum_ means special dvc file with no dependencies, considered
```
DVC-file

shcheklein replied:
is it `always checksum` or something else? cc @efiop

efiop replied:
It is something else, for sure. Probably `always changed`.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,12 @@ differ are listed. For each item listed, either the file name or the checksum is
-  - _changed_ means the named file has changed
+  - _changed deps_ means the named file has changed dependencies
+  - _changed outs_ means the named file has changed outputs
+  - _changed checksum_ means actual check sum of data dosen't match specified in
```
DVC-files

shcheklein replied:
checksum

shcheklein replied:
`of data` -> of the file

dosen't -> doesn't

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,12 @@ differ are listed. For each item listed, either the file name or the checksum is
+  - _changed deps_ means the named file has changed dependencies
+  - _changed outs_ means the named file has changed outputs
+  - _changed checksum_ means actual check sum of data dosen't match specified in
+    dvc files
```
in the DVC-file

sanidhyamangal commented:
@shcheklein, I have successfully updated all the typo's in the description section. Kindly review them now.

efiop commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,12 @@ differ are listed. For each item listed, either the file name or the checksum is
 
 - For the local workspace:
-  - _changed_ means the named file has changed
+  - _changed deps_ means the named file has changed dependencies
```
```suggestion
  - _changed deps_ means the DVC-file has changed dependencies
```

efiop commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,12 @@ differ are listed. For each item listed, either the file name or the checksum is
 - For the local workspace:
-  - _changed_ means the named file has changed
+  - _changed deps_ means the named file has changed dependencies
+  - _changed outs_ means the named file has changed outputs
```
```suggestion
  - _changed outs_ means the DVC-file has changed outputs
```

efiop commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,12 @@ differ are listed. For each item listed, either the file name or the checksum is
-  - _changed_ means the named file has changed
+  - _changed deps_ means the named file has changed dependencies
+  - _changed outs_ means the named file has changed outputs
+  - _changed checksum_ means actual checksum of the file doesn't match specified
```
```suggestion
  - _changed checksum_ means actual checksum of the DVC-file doesn't match the one specified in it
```

efiop commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,12 @@ differ are listed. For each item listed, either the file name or the checksum is
+  - _changed checksum_ means actual checksum of the file doesn't match specified
+    in DVC-files
+  - _always checksum_ means special dvc file with no dependencies, considered
+    always changed
```
```suggestion
```

efiop commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,12 @@ differ are listed. For each item listed, either the file name or the checksum is
+  - _changed outs_ means the named file has changed outputs
+  - _changed checksum_ means actual checksum of the file doesn't match specified
+    in DVC-files
+  - _always checksum_ means special dvc file with no dependencies, considered
```
```suggestion
  - _always changed_ means that this is a special DVC-file with no dependencies, considered
```

efiop commented on static/docs/commands-reference/status.md:
```diff
@@ -118,14 +123,14 @@ cache. For the typical process to update workspaces, see
-                  changed:  foo
+bar.dvc:
+        changed deps:
+                changed:      bar
```
This is an old status. Now we provide more info. See https://github.com/iterative/dvc.org/pull/416/files#r293507445

sanidhyamangal replied:
@efiop  so shall I update this changed with any of these random statuses or do you want any particular one?

efiop replied:
@sanidhyamangal having multiple would be great. Thanks! :slightly_smiling_face:

sanidhyamangal replied:
@efiop  okay cool, I will update them.

sanidhyamangal commented:
@shcheklein I have updated all the requested changes kindly review them.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -7,12 +7,12 @@ cache and remote cache.
 
 positional arguments:
-  targets               DVC files.
+  targets               Limit the scope to these stage files.
```
update it to the new version?

sanidhyamangal replied:
@shcheklein  Yes, updated it to newer version was waiting for original DVC repo to be updated. Since it got approved so pushed these changes now kindly review them

shcheklein replied:
@sanidhyamangal I think the DVC patch has been merged. Feel free to update the docs.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -56,7 +56,16 @@ differ are listed. For each item listed, either the file name or the checksum is
+  - _new_ means the DVC-file has been created but not added into local cache
+  - _modified_ means the dependecies and output is changed in the DVC-file
+  - _deleted_ means the DVC-file has been deleted from local cache
+  - _not in cache_ means the DVC-file no longer exist in local cache
```
so, this is not exactly correct presentation of those states.

first 4 mean that state of the file itself

last 4 mean the state of the output or dependency

We need to change the way we present them.

Please, take the get-started repo, do some simple experiments to get those states, see the pattern.

sanidhyamangal commented:
@shcheklein I have updated all the changes as per your request please review them.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -55,15 +56,32 @@ DVC-file (stage) with differences, the _dependencies_ and/or _outputs_ that
+  - _always changed_ means that this is a special DVC-file with no dependencies,
+    which is considered always changed
+
+- For the output displayed in local workspace:
```
is it the same for dependencies?

sanidhyamangal replied:
Yes, it is the same for dependencies too.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -55,15 +56,32 @@ DVC-file (stage) with differences, the _dependencies_ and/or _outputs_ that
 
-- For the local workspace:
-  - _changed_ means the named file has changed
+- For the changes detected in local workspace:
```
it's not clear how this status is connected with the deps/outs statues below. It looks they are two separate lists which is not the case. One of them encloses the other.

sanidhyamangal replied:
@shcheklein, I have merged these two lists into the previous heading this should work now. Kindly review the changes.

sanidhyamangal commented:
@shcheklein  I have updated all the requested changes and addressed all the questions please review these changes.

sanidhyamangal commented:
@shcheklein it is clear in the para above the list that all the changes in file stages and outputs displayed defining a relationship file-level statuses and output/dependency-level statuses. please read it, in case it's not clear I will rephrase that para.

shcheklein commented:
@sanidhyamangal `changed deps` is not an item status, `modified` is an item status. It's confusing that both come in the same list. It's confusing the way they are described in the para above.

sanidhyamangal commented:
@shcheklein I have embellished the para above the list of local workspace and removed `changed deps` and `changed outs` as it would have been redundant with `modified` prefix creating more confusion. Kindly review these new changes.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,17 +54,30 @@ be rerun if `dvc repro` were executed.
+    one specified in it in DVC-files
+  - _always changed_ means that this is a special DVC-file with no dependencies,
+    which is considered always changed
+  - _new_ output is displayed when a new dependeny or output state is added to
```
remove first output - it's confusing since we use output in two different meanings in the same sentence

shcheklein replied:
same below

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,17 +54,30 @@ be rerun if `dvc repro` were executed.
 - For the local workspace:
-  - _changed_ means the named file has changed
+
+  - _changed checksum_ means actual checksum of the DVC-file doesn't match the
```
can you show an example how does it look like? The same for always changed and changed checksum. Let's artificially create a few example and see how does it look like, please.

there is still some disconnect in this list, in what is described in the paragraph above.

Specifically,  _changed_, _changed checksum_, and _ always changed_ are statuses of the DVC-file itself.

sanidhyamangal replied:
@shcheklein shouldn't we add some examples in simple usage section?

shcheklein replied:
that would be ideal, sure!

sanidhyamangal replied:
@shcheklein I have added the examples in the simple usage section kindly check them out.

shcheklein replied:
this is cool. So, you see:

```
test.dvc:
	changed outs:
		modified:           test
	changed checksum
	always changed
```

Items like `modified` are actually not on the same level of hierarchy as `always changed`.

Why don't we come with a better description that explains these two lists separately. And explains the connection between them.

sanidhyamangal commented:
@shcheklein  I have updated the requested changes kindly go through them.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,17 +54,30 @@ be run again if `dvc repro` were executed.
 differ are listed. For each item listed, either the file name or the checksum is
-shown, and additionally a status word is shown describing the change:
+shown, and additionally, a status word is shown describing the changes in
+checksum or status for both local and cloud workspace. Given list describes
```
Given list sounds like a reference to the listed items above, not to the list below

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,17 +54,30 @@ be run again if `dvc repro` were executed.
+  - _always changed_ means that this is a special DVC-file with no dependencies,
+    which is considered always changed
+  - _new_ is displayed when a new dependeny or output state is added to DVC-file
+  - _modified_ is displayed when any dependencies or output states is modified
```
dependencies ... are
dependency ... is

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,17 +54,30 @@ be run again if `dvc repro` were executed.
+  - _new_ is displayed when a new dependeny or output state is added to DVC-file
+  - _modified_ is displayed when any dependencies or output states is modified
+    in DVC-file
+  - _deleted_ is displayed when any exsisting dependencies or output states are
```
same here

sanidhyamangal commented:
@shcheklein I have split the list into hierarchical format now as per our discussion, I hope this provides a better separability. In case you don't find it that appealing then kindly suggest me a different method.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,19 +53,40 @@ This says that no differences were detected, and therefore that no stages would
+    which is considered always changed
+  - _changed deps_ means that there are some changes in dependencies that are
+    incorporated in the DVC-file, some of the prominent states are:
+    - _new_ : when new dependencies are added into DVC-file
```
in this and the following items: it's more about the state of a file or a directory that is dependency, not about changes in the DVC-file. Let's discuss this on Discrord.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,19 +53,40 @@ This says that no differences were detected, and therefore that no stages would
+  - _always changed_ means that this is a special DVC-file with no dependencies,
+    which is considered always changed
+  - _changed deps_ means that there are some changes in dependencies that are
+    incorporated in the DVC-file, some of the prominent states are:
```
please, remove prominent. It's a complete list, no need to put extra words on top

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,19 +53,40 @@ This says that no differences were detected, and therefore that no stages would
-  - _changed_ means the file has changed
+
+  - _changed checksum_ means actual checksum of the DVC-file doesn't match the
+    one specified in it in DVC-files
```
fix "in it in"

jorgeorpinel replied:
"means that the update checksum of the data doesn't match the one saved in the DVC-files."

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,19 +53,40 @@ This says that no differences were detected, and therefore that no stages would
+    - _new_ : when new dependencies are added into DVC-file
+    - _modified_ : when an exsisting dependencies are modified in DVC-file
+    - _deleted_ : when an exsisting dependencies are removed from DVC-file
+    - _not in cache_ : when dependencies mentioned in DVC-file no longer exsists
```
no longer - it can be not yet

let's just keep "... don't exist in DVC cache"

shcheklein replied:
mind the comment above ^^

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,19 +53,40 @@ This says that no differences were detected, and therefore that no stages would
+
+  - _changed checksum_ means actual checksum of the DVC-file doesn't match the
+    one specified in it in DVC-files
+  - _always changed_ means that this is a special DVC-file with no dependencies,
```
for all the items, there should be some rule about punctuation at the end. Most likely ; will do. @jorgeorpinel do you know about this?

jorgeorpinel replied:
I haven't standardized this yet but actually I think we mostly use `.` throughout the docs since our bullet lists tend to contain long paragraphs.

jorgeorpinel replied:
Also on this line, note that "special DVC-file with no dependencies" is also known as an "orphan" stage file, as mentioned in a few docs:

![image](https://user-images.githubusercontent.com/1477535/60850320-09a89200-a1ac-11e9-837f-4d9ceb17136d.png)

shcheklein replied:
good points. I think the rule should be - if items are short one sentence long, and start from lowercase then we should be using `;` (or nothing?), otherwise `.` make sense.

jorgeorpinel replied:
OK, I just standardized it in 3134296. Long list items (paragraphs) and all cmd ref/ config option list items end in period `.` and the list items are separated by a new line. Shorter lists (a sentence or less) end in semicolon `;` and don't have extra new lines in between. I also fixed the indentation of a bunch of lists.

The part about starting in lower case varies though. All the cmd ref option list items start in lower case even when they tend to be full paragraphs for ex., since they are preceded by a `-` dash. E.g.

```
- `-q`, `--quiet` - do not write anything to standard output. Exit with 0 if no
  problems arise, otherwise 1.
```

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -53,19 +53,40 @@ This says that no differences were detected, and therefore that no stages would
+_outputs_ that differ are listed. For each item listed, either the file name or
+the checksum is shown, and additionally, a status word is shown describing the
+changes in checksum or status for both local and cloud workspace. Given list
+provides a reference to both, the status of the DVC-file and differences in the
```
`the status of the DVC-file as well as dependencies and outputs:`

jorgeorpinel replied:
What is a "cloud workspace"?
These last 2 sentences in the paragraph are confusing. They seem unnecessarily long and maybe repetitive. Feel free to break up the paragraph in 2 if needed or even re-write the whole thing from scratch, as long as the explanation is clear.

jorgeorpinel replied:
Moved to https://github.com/iterative/dvc.org/pull/416/files#r303261384

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -7,8 +7,8 @@ cache and remote cache.
-                  [-r REMOTE] [-a] [-T] [-d]
+usage: dvc status [-h] [-v] [-j JOBS] [--show-checksums]
+                  [-q] [-c] [-r REMOTE] [-a] [-T] [-d]
                   [targets [targets ...]]
```
Hi. Why was this changed? In fact it can all fit in 2 lines:
```console
usage: dvc status [-h] [-v] [-j JOBS] [--show-checksums] [-q] [-c]
                  [-r REMOTE] [-a] [-T] [-d] [targets [targets ...]]
```

jorgeorpinel replied:
p.s. I updated this in 5709d16 which is a change in my current PR, #464 so you can ignore this one but will probably cause a conflict before you can merge... Or feel free to also apply the same change in advance to avoid such Git conflict.

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -53,19 +53,40 @@ This says that no differences were detected, and therefore that no stages would
 be run again if `dvc repro` were executed.
```
Please improve grammar to: "...no stages would be run again by `dvc repro`."

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -53,19 +53,40 @@ This says that no differences were detected, and therefore that no stages would
-DVC-file (stage) with differences, the _dependencies_ and/or _outputs_ that
-differ are listed. For each item listed, either the file name or the checksum is
-shown, and additionally a status word is shown describing the change:
+DVC-file (stage) with differences, the changes in both _dependencies_ and/or
```
Can't use "both" with "and/or". I'd remove "both" since "and/or" is more correct.

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -53,19 +53,40 @@ This says that no differences were detected, and therefore that no stages would
-shown, and additionally a status word is shown describing the change:
+DVC-file (stage) with differences, the changes in both _dependencies_ and/or
+_outputs_ that differ are listed. For each item listed, either the file name or
+the checksum is shown, and additionally, a status word is shown describing the
```
I don't see the need for that extra comma after `and additionally`, consider reverting.

sanidhyamangal commented:
@shcheklein Kindly review the updates, I have added punctuations in the end, some more insights about the list as per our discussion of _discord_ and fixed some issues suggested by @jorgeorpinel.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -50,21 +50,45 @@ detected, `dvc status` prints this message:
+      not reflected in the DVC-file;
+    - _deleted_ : when an exsisting outputs are removed from the workspace but
+      still referred in the DVC-file;
+    - _not in cache_ : when outputs mentioned in DVC-file no longer exsists in
```
exists -> exist

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -50,21 +50,45 @@ detected, `dvc status` prints this message:
+    (orphans), which is considered always changed;
+  - _changed deps_ means that there are some changes in dependencies that are
+    incorporated in the DVC-file, some of the states are:
+    - _new_ : when reference of new dependency files are added into DVC-file;
```
@efiop just to double check. is it a correct description of this status? can you take a brief look at other statuses as well please.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -50,21 +50,45 @@ detected, `dvc status` prints this message:
-  - _changed_ means the file has changed
+
+  - _changed checksum_ means actual checksum of the files present in workspace
+    doesn't match the one saved in DVC-files;
```
it deserves a better explanation tbh. Also, `files` -> `the file`, `in DVC-files` -> `in the DVC-file` ... no need to mention workspace here. The rule of thumb - would you understand anything from this description if you were reading this first time?

jorgeorpinel replied:
How about "means that the update checksum of the data doesn't match the one saved in the DVC-files" as I suggested in https://github.com/iterative/dvc.org/pull/416#pullrequestreview-257759178?

sanidhyamangal replied:
updated it in my latest commit, kindly review them.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -50,21 +50,45 @@ detected, `dvc status` prints this message:
+_outputs_ that differ are listed. For each item listed, either the file name or
+the checksum is shown, and additionally a status word is shown describing the
+changes in checksum or status. Given list provides a reference to both, the
+status of the DVC-file and differences in the DVC-files:
```
`differences in the DVC-files` -> `and changes to individual dependencies and outputs described by the file`

`status of the DVC-file` -> `status of a DVC-file`

jorgeorpinel replied:
Here, full suggestion for the last 2 sentences:

"For each item listed, either the file name or the checksum is shown, and additionally a status word is shown describing the changes. This changes list provides a reference to both the status of a DVC-file, as well as the changes to individual dependencies and outputs described in it:"

sanidhyamangal replied:
revised the para based on these suggestions, kindly check them.

jorgeorpinel replied:
Yes, thanks. But actually we should also end the paragraph in period `.` please (since there are columns `:` in the main bullets below. (Like I mentioned in https://github.com/iterative/dvc.org/pull/416#discussion_r307958841)

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -50,21 +50,45 @@ detected, `dvc status` prints this message:
+  - _changed outs_ means that there are some changes in output states that are
+    incorporated in the DVC-file, some of the states are:
+    - _new_ : when reference of new output files are are added into DVC-file;
+    - _modified_ : when md5 of an exsisting outputs are changed in workspace but
```
if you mention checksum everywhere above, let's use checksum here as well, instead of md5

sanidhyamangal replied:
Done!

shcheklein commented on static/docs/commands-reference/status.md:
````diff
@@ -7,8 +7,8 @@ cache and remote cache.
 ```usage
-usage: dvc status [-h] [-v] [-j JOBS] [--show-checksums] [-q] [-c]
-                  [-r REMOTE] [-a] [-T] [-d]
+usage: dvc status [-h] [-v] [-j JOBS] [--show-checksums]
````
please, address the @jorgeorpinel 's comment ... no split is required

jorgeorpinel replied:
Exact txt below 🙂 

```
usage: dvc status [-h] [-v] [-j JOBS] [--show-checksums] [-q] [-c]
                  [-r REMOTE] [-a] [-T] [-d] [targets [targets ...]]
```

sanidhyamangal replied:
reverted it to the original format, unsplit the list. Please review it.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -50,21 +50,45 @@ detected, `dvc status` prints this message:
+  - _always changed_ means that this is a special DVC-file with no dependencies
+    (orphans), which is considered always changed;
+  - _changed deps_ means that there are some changes in dependencies that are
+    incorporated in the DVC-file, some of the states are:
```
`some` - it's not `some`, it should be a complete list

sanidhyamangal replied:
Paraphrased the given line, please review them.

shcheklein commented on static/docs/commands-reference/status.md:
```diff
@@ -50,21 +50,45 @@ detected, `dvc status` prints this message:
+    - _not in cache_ : when dependencies mentioned in DVC-file no longer exsists
+      in local cache;
+  - _changed outs_ means that there are some changes in output states that are
+    incorporated in the DVC-file, some of the states are:
```
`some` - not some, it should be a complete list

sanidhyamangal replied:
Paraphrased for this line as well kindly review the changes.

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -50,21 +50,45 @@ detected, `dvc status` prints this message:
+  - _changed checksum_ means actual checksum of the files present in workspace
+    doesn't match the one saved in DVC-files;
+  - _always changed_ means that this is a special DVC-file with no dependencies
+    (orphans), which is considered always changed;
```
Note that "special DVC-file with no dependencies" is also known as an "orphan" stage file, as mentioned in a few docs:

![image](https://user-images.githubusercontent.com/1477535/60850320-09a89200-a1ac-11e9-837f-4d9ceb17136d.png)

Should we mention that term here? (@shcheklein)

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -50,21 +50,45 @@ detected, `dvc status` prints this message:
 
 This says that no differences were detected, and therefore that no stages would
-be run again if `dvc repro` were executed.
+be run again by `dvc repro`.
```
- [ ] `This says` -> `This indicates`.
- [ ] `therefore that no` -> `therefore no`

sanidhyamangal replied:
amended, kindly review these changes.

sanidhyamangal commented:
@shcheklein I have addressed all the issues and reformed the file based on requested changes. Kindly review the updated version.

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -48,22 +48,47 @@ detected, `dvc status` prints this message:
+the checksum is shown, and additionally a status word is shown describing the
+changes. This changes list provides a reference to both the status of a
+DVC-file, as well as the changes to individual dependencies and outputs
+described in it:
```
Please end this paragraph in period `.` not column `:`

jorgeorpinel replied:
Fixed in d0bbaba

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -48,22 +48,47 @@ detected, `dvc status` prints this message:
+DVC-file, as well as the changes to individual dependencies and outputs
+described in it:
 
 - For the local workspace:
```
This shouldn't be a bullet with sub-bullets. How about using a `###` header? Or just a **bold** phrase.

jorgeorpinel replied:
...and un-indent the bullets below, of course 🙂

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -48,22 +48,47 @@ detected, `dvc status` prints this message:
+    - _not in cache_ : when outputs mentioned in DVC-file no longer exsist in
+      local cache;
+
 - For comparison against a remote cache:
```
Same as https://github.com/iterative/dvc.org/pull/416#discussion_r307958847

jorgeorpinel commented on static/docs/commands-reference/status.md:
```diff
@@ -48,22 +48,47 @@ detected, `dvc status` prints this message:
-For the _changed_ case, the `dvc repro` command is indicated.
+  - _new_ means the file exists in the local cache but not the remote cache;
+  - _deleted_ means the file doesn't exist in the local cache, but exists in the
+    remote cache;
```
In all of the bullets above (starting in line 64), use consistent formatting:

- [x] All bullets already start with _italic_ term
- [ ] All bullets should either have column `: ` after the term OR the word ` means `, and then the description. Actually I vote for ` - ` instead, like in all the list of the existing command references.
- [ ] All bullets should be full sentences and end in period `.`  We no longer use semicolons `;` for this.

Thanks

jorgeorpinel replied:
Also fixed in d0bbaba
